packages/ai-openai/src/node/openai-language-models-manager-impl.ts

```diff
@@ -33,19 +33,34 @@ export class OpenAiLanguageModelsManagerImpl implements OpenAiLanguageModelsMana
 
     // Triggered from frontend. In case you want to use the models on the backend
     // without a frontend then call this yourself
-    async createLanguageModels(...modelIds: string[]): Promise<void> {
-        for (const id of modelIds) {
-            // we might be called by multiple frontends, therefore check whether a model actually needs to be created
-            if (!(await this.languageModelRegistry.getLanguageModel(`openai/${id}`))) {
-                this.languageModelRegistry.addLanguageModels([new OpenAiModel(id, () => this.apiKey)]);
+    async createOrUpdateLanguageModels(...modelDescriptions: OpenAiModelDescription[]): Promise<void> {
+        for (const modelDescription of modelDescriptions) {
+            const model = await this.languageModelRegistry.getLanguageModel(modelDescription.id);
+            if (model) {
+                if (!(model instanceof OpenAiModel)) {
+                    console.warn(`Open AI: model ${modelDescription.id} is not an OpenAI model`);
+                    continue;
+                }
+                if (!modelDescription.url) {
+                    // This seems to be an official model, but it was already created. This can happen during the initializing of more than one frontends.
```

Comment:
```suggestion
                    // This seems to be an official model, but it was already created. This can happen during the initializing of more than one frontend.
```